IdentityCore/src/oauth2/aad_base/MSIDAADTokenResponse.m

```diff
@@ -22,136 +22,82 @@
+#import "MSIDTokenResponse+Internal.h"
 #import "MSIDTelemetryEventStrings.h"
-#import "MSIDAADV1IdTokenClaims.h"
-#import "MSIDHelpers.h"
-#import "MSIDRefreshableToken.h"
-#import "MSIDBaseToken.h"
-
-@interface MSIDAADTokenResponse ()
-
-@property (readonly) NSString *rawClientInfo;
-
-@end
 
 @implementation MSIDAADTokenResponse
 
-// Default properties for an error response
-MSID_JSON_ACCESSOR(MSID_OAUTH2_CORRELATION_ID_RESPONSE, correlationId)
-
-// Default properties for a successful response
-MSID_JSON_ACCESSOR(MSID_OAUTH2_RESOURCE, resource)
-MSID_JSON_RW(MSID_OAUTH2_CLIENT_INFO, rawClientInfo, setRawClientInfo)
-MSID_JSON_ACCESSOR(MSID_FAMILY_ID, familyId)
-MSID_JSON_ACCESSOR(MSID_TELEMETRY_KEY_SPE_INFO, speInfo)
-MSID_JSON_ACCESSOR(MSID_OAUTH2_SUB_ERROR, suberror)
-MSID_JSON_ACCESSOR(@"adi", additionalUserId)
-
-- (instancetype)initWithJSONDictionary:(NSDictionary *)json
-                          refreshToken:(MSIDBaseToken<MSIDRefreshableToken> *)token
-                                 error:(NSError **)error
+- (instancetype)initWithAccessToken:(NSString *)accessToken
+                       refreshToken:(NSString *)refreshToken
+                          expiresIn:(NSInteger)expiresIn
+                          expiresOn:(NSInteger)expiresOn
+                  extendedExpiresIn:(NSInteger)extendedExpiresIn
+                  extendedExpiresOn:(NSInteger)extendedExpiresOn
+                          tokenType:(NSString *)tokenType
+                              scope:(NSString *)scope
+                              state:(NSString *)state
+                            idToken:(NSString *)idToken
+               additionalServerInfo:(NSDictionary *)additionalServerInfo
+                              error:(NSString *)error
+                           suberror:(NSString *)suberror
+                   errorDescription:(NSString *)errorDescription
+                         clientInfo:(MSIDClientInfo *)clientInfo
+                           familyId:(NSString *)familyId
+                   additionalUserId:(NSString *)additionalUserId
+                            speInfo:(NSString *)speInfo
+                      correlationId:(NSString *)correlationId
+                          initError:(NSError **)initError
 {
-    self = [super initWithJSONDictionary:json refreshToken:token error:error];
+    self = [super initWithAccessToken:accessToken
+                         refreshToken:refreshToken
+                            expiresIn:expiresIn
+                            tokenType:tokenType
+                                scope:scope
+                                state:state
+                              idToken:idToken
+                 additionalServerInfo:additionalServerInfo
+                                error:error
+                     errorDescription:errorDescription
+                            initError:initError];
     
     if (self)
     {
-        [self initDerivedProperties];
+        _expiresOn = expiresOn;
```

Comment:
same here, verify the fields you receive.

Reply:
Agreed to not verify them at this point.